crates/lox_core/src/bodies.rs

```diff
@@ -62,8 +64,223 @@ pub fn gravitational_parameter<T: PointMass>(_: T) -> f64 {
     <T as PointMass>::gravitational_parameter()
 }
 
+const N_COEFFICIENTS: usize = 18;
+
+type PolynomialCoefficients = (f64, f64, f64, [f64; N_COEFFICIENTS]);
+type NutationPrecessionCoefficients = ([f64; N_COEFFICIENTS], [f64; N_COEFFICIENTS]);
+
+pub trait RotationalElements {
+    fn nutation_precession_coefficients() -> NutationPrecessionCoefficients;
+    fn right_ascension_coefficients() -> PolynomialCoefficients;
+
+    fn declination_coefficients() -> PolynomialCoefficients;
+
+    fn prime_meridian_coefficients() -> PolynomialCoefficients;
+
+    fn theta(t: f64) -> [f64; N_COEFFICIENTS] {
```

Comment:
Do you need a trait if none of these are methods (ie. they don't reference self)?

Reply:
Yes. The celestial bodies are modelled as empty structs and this allows us to attach data to them.